### DISCUSSION TITLE
CIP 38 - Granda Mento - Discussions

### DISCUSSION THREAD

**Discussion by tkporter** [celo-org/celo-proposals#341] [CIP Discussions]
Discussions for CIP 38, Granda Mento

Relevant previous discussions:
* PR comments on initial proposal: https://github.com/celo-org/celo-proposals/pull/224/
* Forum post with initial idea including high level discussions and debate around the approach Granda Mento makes: https://forum.celo.org/t/discussion-on-granda-mento-enabling-larger-stablecoin-mints/966

**Comment by tkporter** [comment]
Some relevant topics for discussion:

* There was some advocacy for proposers providing the rate they wish to exchange at that must be within X% of the oracle rate. The current proposal uses the oracle rate directly, which was the product of rough consensus. If you feel strongly about the proposers including the price they want, feel free to mention here.
* The CIP doesn't include a concrete process of what is expected of an exchange proposal. See [these comments](https://github.com/celo-org/celo-proposals/pull/224#discussion_r639754818) for some ideas.
* Dealing with StableToken demurrage - the current proposal stores `sellAmount` when a stable token is being sold in terms of units. Another workaround is to have each exchange proposal create a new lightweight contract where the stable token is deposited, that way the entire balance of the deposit after demurrage is clear.

Note this CIP doesn't seek to define parameters like the spread, exchange limits, etc-- this will be discussed in the CGP to activate Granda Mento down the road.

**Comment by tkporter** [comment]
@yorhodes suggested allowing the approver multisig to cancel exchange proposals when they are in the `Approved` state.

Part of the reasoning why only Governance can cancel in the Approved state was to prevent the exchanger from having a "free option" where they can just cancel the exchange when the price they locked in at proposal-time is worse than the price at exchange-execution-time. Giving the approver the ability to cancel when it's Approved could make it easier for the exchanger to do this if they're able to convince the approver to cancel it on their behalf. But allowing this could prevent needing to rally the community behind a governance proposal if the approvers mistakenly approved a proposal. Generally I'm supportive of allowing this, but I'm curious to hear any thoughts.

**Comment by asaj** [comment]
My only concern is that exchanges may seek to influence the approvers to cancel if the price moves out from under them significantly, and that the approvers may be susceptible to this influence. Given the approver is likely to be a multisig, I really doubt anything is likely to get approved "accidentally", and so I'd be curious if there are other circumstances in which this feature would be useful

**Comment by martinvol** [comment]
If the exchanger has the option to cancel at any time before the execution, then nothing prevent's the exchangers from cancelling until their trade is favorable. As governance is very unlikely to cancel trades with relatively small price movements, that'd would leave an exchanger that wants to use Granda Mento solely for trading against the reserve with positive expected value return. So, I'd not change what is stated in the CIP.

**Comment by tkporter** [comment]
Both good points, I agree that the exchanger will probably try to influence the approvers & that it's very unlikely approvers would approve something and want to immediately take it back. Going to continue with the current design.
